Bottom: 97e68a2b75530d4fa7e09ec9d204f890b39bbef7
Top:    752ca4c77a41eb6b8a5bd4537b853d19b2811560
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-26 22:04:30 +0000

Fix usage of hash tables - don't free the inserted key!


---

diff --git a/src/hid/step/step_writer.c b/src/hid/step/step_writer.c
index 3ec1b4b..acec485 100644
--- a/src/hid/step/step_writer.c
+++ b/src/hid/step/step_writer.c
@@ -209,7 +209,6 @@ step_cartesian_point (step_file *file, char *name, double x, double y, double z)
 
   g_hash_table_insert (file->cartesian_point_hash, line, GINT_TO_POINTER (file->next_id));
   fprintf (file->f, "#%i=CARTESIAN_POINT(%s);\n", file->next_id, line);
-  g_free (line);
 
   return file->next_id++;
 }
@@ -230,7 +229,6 @@ step_direction (step_file *file, char *name, double x, double y, double z)
 
   g_hash_table_insert (file->direction_hash, line, GINT_TO_POINTER (file->next_id));
   fprintf (file->f, "#%i=DIRECTION(%s);\n", file->next_id, line);
-  g_free (line);
 
   return file->next_id++;
 }
@@ -283,7 +281,6 @@ step_vector (step_file *file, char *name, step_id orientation, double magnitude)
 
   g_hash_table_insert (file->vector_hash, line, GINT_TO_POINTER (file->next_id));
   fprintf (file->f, "#%i=VECTOR(%s);\n", file->next_id, line);
-  g_free (line);
 
   return file->next_id++;
 }
